designs/default-zero-value.md

```diff
@@ -0,0 +1,872 @@
+   replaced with the `@default` trait.
+
+
+### Balancing old and new assumptions
+
+Smithy comes from a long line of service frameworks built inside Amazon dating
+back nearly 20 years. Until this proposal, the required trait was only treated
+as server-side input validation. This causes two major issues that impacts the
+AWS SDK team's ability to fully use the required trait for code generation:
+
+1. Many teams did not consider the required trait as something that impacts
+   output. This means that there are many members in AWS that should be
+   marked as required but are not, and there are members that are required in
+   input, but not always present in output.
+2. Most teams designed their APIs under the guidance that the required trait
+   can be removed without breaking their end users.
+
+We will work to improve the accuracy and consistency of the required trait in
+AWS API models where possible, but it will take significant time and effort
+from hundreds of different teams within AWS. To accommodate this shift in
+modeling changes, we will introduce AWS-specific traits to influence code
+generation and provide AWS SDK specific code generation recommendations.
+
+Some of this guidance will admittedly be an inconvenience for developers, but
+it is less catastrophic than previously generated client code failing at
+runtime when deserializing the response of an operation, and it matches the
+current reality of how AWS was modeled. While it's true client code might
+start failing anyways if a service stops sending output members that were
+previously marked as required, it would only fail when the client explicitly
+attempts to dereference the value, and this kind of change is generally only
+made when a client opts-into some new kind of functionality or workflow. If an
+SDK fails during deserialization because a required output member is missing,
+then customer applications would be completely broken until they update their
+SDK.
+
+
+#### Special casing top-level input structures
+
+AWS SDKs SHOULD generate every required top-level input member as optional even
+if they're marked as required (that is, the members of structures marked with
+the `@input` trait). This gives AWS service teams the ability to remove the
+required trait from top-level input members without risking breaking previously
+generated clients. If such a change would not break previously generated
+clients for a given AWS SDK, then it is free to ignore this guidance.
+
+Nearly 80% of all required traits used in AWS today appear in only top-level
+members of operation inputs (~24,500 total required members vs ~19,000
+required top-level input members). By special casing top-level operation input
+members as optional even if they are marked with the required trait, this
+proposal becomes significantly safer for AWS SDKs to adopt.
+
+
+#### `aws.api#serverRequired` trait
+
+The `aws.api#serverRequired` trait is used instead of `required` to indicate
```

Comment:
Does this mean that AWS specifications will see a transition of trait from `smithy.api#required` to `aws.api#serverRequired` ? That'd be really nice for third parties